### ISSUE TITLE
Can't unpin a site that redirects on load

### ISSUE THREAD

**Issue by petemill** [brave/browser-laptop#13816]
If you have a pinned site in session-store (`[URL]`)that when loaded redirects to `[URL/x]` immediately on load, then the user will never be able to un-pin the site.

This is due to comparing the key based on the current location of the tab.

I had this happen with `https://inbox.google.com/` pinned, which always redirects to `https://inbox.google.com/b/0`

We should probably compare to the initial provisional location of the tab, i.e. the one sent to create the tab from the pinnedSite data.

Reproducible on 0.22.13 (current release), beta and master.

**Comment by HunterMeyer** [comment]
@petemill I am experiencing this same issue. For me, the URL was for Google Sheets which added a hash to the end of the URL when I loaded the page.

**I was able to get the "Unpin" tab to function by:**
**1.** turn off JavaScript in Brave's settings
**2.** Open a new Brave window and click "Unpin" on the tab
![block-scripts](https://user-images.githubusercontent.com/4690516/40924562-894590dc-67e5-11e8-9570-905022edd670.png)

Another thing that might be helpful to slow or prevent the redirect is to set the network speed to a slow 36 setting. You can do this by opening the developer console (View > Toggle Developer Tools) then select an option from the network menu. 
![slow](https://user-images.githubusercontent.com/4690516/40924776-1babc3e2-67e6-11e8-87b7-b34724978185.gif)

This is a bug nonetheless. But this workaround saved me from abandoning Brave.

**Comment by lauroaraujo** [comment]
I had a similar issue, AWS console page pinned while logged in. After a few days I tried to unpin it but it would always return.

After seeing this thread I tried disabling javascript, which did not work. Then I tried logging into AWS and then unpinning, and this time it worked. I still had javascript disabled, don't know if it was needed or not.
 
Brave: 0.23.31 
V8: 6.7.288.46 
rev: 3148acef36dba0fce89108638bb27927c4937f90 
Muon: 7.1.5 
OS Release: 4.15.0-24-generic 
Update Channel: Release 
OS Architecture: x64 
OS Platform: Linux 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 67.0.3396.103
